### PR TITLE
Fix reading java array type

### PR DESCRIPTION
**Pull request by KasiaKoz**
Fixes https://github.com/arup-group/genet/issues/220
The java array types were being read to a set with extra unnecessary characters.

E.g. given: 
```xml
<attribute name="attrib" class="java.lang.Array">{'A', 'B'}</attribute>
```
would be read to
```python
{"{'A'", " 'B'}"}
```
This is now fixed to give:
```python
{"A", "B"}
```

Added a test to catch the problem:
Screenshot 2023-12-18 at 18 10 04

### CODE REVIEW

**Review by mfitz — COMMENTED** [review]
The bug fix looks good, but I've queried the conversion from list to set - are we okay with losing duplicate elements? Can this cause a problem?

File: tests/test_input_matsim_reader.py [hunk 1]
```diff
@@ -1038,3 +1038,25 @@ def make_network_with_elevations_xml_string(nodes_with_elevations_dict, link_dic
...
+        self.text = text
+        self.attrib = {"name": "doesn't matter"}
+
+
+@pytest.mark.parametrize(
+    "case,input_elem_text,expected_output",
+    [
+        ("is_just_a_string", "hello", "hello"),
+        ("is_a_java_array", "{'hello', 'yes'}", {"hello", "yes"}),
+        ("is_none", None, ""),
```
[the remark this answers is not in the document]
Comment (KasiaKoz) [reply tests/test_input_matsim_reader.py:1054]: yeah they live as set so no duplicates and order is lost, so far not a problem but up for debate

**Comment by KasiaKoz** [comment]
Made an issue for the bug and updated the PR description

**Review by mfitz — APPROVED** [review]
Cool.

**Pull request merged**